[hipblasLT] SK use Tree reduction if < 2 split (#3403)
ROCm/rocm-libraries: projects/hipblaslt/CHANGELOG.md

Full documentation for hipBLASLt is available at [rocm.docs.amd.com/projects/hipBLASLt](https://rocm.docs.amd.com/projects/hipBLASLt/en/latest/index.html).

## hipBLASLt 1.2.1 for ROCm 7.2.1

### Resolved issues

> * Fix issue where users may encounter HIPBLAS_STATUS_INTERNAL_ERROR with various sizes in CPX mode

Reviewer: ```suggestion
* Fix issue where users might encounter a `HIPBLAS_STATUS_INTERNAL_ERROR` with various sizes in CPX mode.
```
